core/eolearn/tests/test_core_tasks.py

```diff
@@ -71,38 +73,34 @@ def patch_fixture():
     return patch
 
 
-def test_copy(patch):
-    patch_copy = CopyTask().execute(patch)
-
-    assert patch == patch_copy, "Copied patch is different"
-
-    patch_copy.data["new"] = np.arange(1).reshape(1, 1, 1, 1)
-    assert "new" not in patch.data, "Dictionary of features was not copied"
+@pytest.mark.parametrize("task", [DeepCopyTask, CopyTask])
+def test_copy(task: CopyTask, patch: EOPatch) -> None:
+    patch_copy = task().execute(patch)
+    assert deep_eq(patch_copy, patch)
 
     patch_copy.data["bands"][0, 0, 0, 0] += 1
-    assert np.array_equal(patch.data["bands"], patch_copy.data["bands"]), "Data should not be copied"
-
-
-def test_deepcopy(patch):
-    patch_deepcopy = DeepCopyTask().execute(patch)
-
-    assert patch == patch_deepcopy, "Deep copied patch is different"
+    assert not deep_eq(patch_copy, patch) if task == DeepCopyTask else deep_eq(patch_copy, patch)
 
-    patch_deepcopy.data["new"] = np.arange(1).reshape(1, 1, 1, 1)
-    assert "new" not in patch.data, "Dictionary of features was not copied"
+    patch_copy.data["new"] = np.arange(1).reshape(1, 1, 1, 1)
+    assert "new" not in patch.data
 
-    patch_deepcopy.data["bands"][0, 0, 0, 0] += 1
-    assert not np.array_equal(patch.data["bands"], patch_deepcopy.data["bands"]), "Data should be copied"
 
+@pytest.fixture(name="expected_patch")
+def expected_patch_fixture(patch, request) -> EOPatch:
+    return EOPatch(**{feature: getattr(patch, feature) for feature in request.param})
 
-def test_partial_copy(patch):
-    partial_copy = DeepCopyTask(features=[(FeatureType.MASK_TIMELESS, "mask"), FeatureType.BBOX]).execute(patch)
-    expected_patch = EOPatch(mask_timeless=patch.mask_timeless, bbox=patch.bbox)
-    assert partial_copy == expected_patch, "Partial copying was not successful"
 
-    partial_deepcopy = DeepCopyTask(features=[FeatureType.TIMESTAMP, (FeatureType.SCALAR, "values")]).execute(patch)
-    expected_patch = EOPatch(scalar=patch.scalar, timestamp=patch.timestamp)
-    assert partial_deepcopy == expected_patch, "Partial deep copying was not successful"
+@pytest.mark.parametrize(
+    "features, expected_patch",
+    [
+        ([(FeatureType.MASK_TIMELESS, "mask"), FeatureType.BBOX], ["mask_timeless", "bbox"]),
+        ([FeatureType.TIMESTAMP, (FeatureType.SCALAR, "values"), FeatureType.BBOX], ["scalar", "timestamp", "bbox"]),
+    ],
+    indirect=["expected_patch"],
+)
+def test_partial_copy(features: List[FeatureSpec], patch: EOPatch, expected_patch: EOPatch) -> None:
+    partial_copy = DeepCopyTask(features=features).execute(patch)
+    assert deep_eq(partial_copy, expected_patch)
```

Comment:
this is nearly 10 lines of extra code and also less readable :thinking:

Reply:
Planing to use this fixture more other tests, if there would be like 10 test cases it would be less code mark.parametrize is boilerplate code heavy in this case still prefer having two DRY test cases instead of "long" test.

Reply:
speaking of DRY, you are repeating yourself in every test case :)
it also seems like there could be a problem, the way you constructed your `expected_patch_fixture` the whole `data_timeless` gets copied. Is there only a single feature in there? If so, select a different feature type, one with multiple features.